default/python/AI/savegame_codec/_decoder.py

```diff
@@ -11,81 +11,41 @@
 the __dict__ content will be set directly. It is the responsiblity of the trusted classes
 to provide a __setstate__ method to verify and possibly sanitize the content of the passed state.
 """
+import binascii
 import json
 
 import EnumsAI
 from freeorion_tools import profile
-from logging import debug
 
 from ._definitions import (ENUM_PREFIX, FALSE, FLOAT_PREFIX, INT_PREFIX, InvalidSaveGameException, NONE, PLACEHOLDER,
                            SET_PREFIX, TRUE, TUPLE_PREFIX, trusted_classes, )
 
 
-def _ensure_binary(s, encoding='utf-8', errors='strict'):
-    """Coerce **s** to bytes.
-      - `str` -> encoded to `bytes`
-      - `bytes` -> `bytes`
+class SaveDecompressException(Exception):
     """
-    if isinstance(s, str):
-        return s.encode(encoding, errors)
-    elif isinstance(s, bytes):
-        return s
-    else:
-        raise TypeError("not expecting type '%s'" % type(s))
-
-
-def _ensure_str(s, encoding='utf-8', errors='strict'):
-    """Coerce *s* to `str`.
-      - `str` -> `str`
-      - `bytes` -> decoded to `str`
+    Exception class for troubles with decompressing save game string.
     """
-    if not isinstance(s, (str, bytes)):
-        raise TypeError("not expecting type '%s'" % type(s))
-    elif isinstance(s, bytes):
-        s = s.decode(encoding, errors)
-    return s
+    pass
 
 
 @profile
 def load_savegame_string(string: str):
     """
     :rtype: AIstate
+    :raises: SaveUncompressException, InvalidSaveGameException
     """
     import base64
     import zlib
 
-    new_string = string
     try:
         new_string = base64.b64decode(string.encode("utf-8"))
```

Comment:
the encode seems unnecessary here. ```b64decode``` allows to pass a ```str``` and will  use ```encode('ascii')``` internally (which the string  must comply with if it is indeed base64 encoded).